tests/experiments_tests/test_evaluator.py

```diff
@@ -99,52 +118,100 @@ def test_evaluate_if_necessary(self):
-            n_runs=self.n_runs,
+        agent_evaluator = evaluator.AsyncEvaluator(
+            n_runs=self.n_episodes,
             eval_interval=3,
             outdir=outdir,
             max_episode_len=None,
             step_offset=0,
             save_best_so_far_agent=self.save_best_so_far_agent,
         )
 
-        evaluator.evaluate_if_necessary(t=1, episodes=1, env=env, agent=agent)
+        agent_evaluator.evaluate_if_necessary(
+            t=1, episodes=1, env=env, agent=agent)
         self.assertEqual(agent.act.call_count, 0)
 
-        evaluator.evaluate_if_necessary(t=2, episodes=2, env=env, agent=agent)
+        agent_evaluator.evaluate_if_necessary(
+            t=2, episodes=2, env=env, agent=agent)
         self.assertEqual(agent.act.call_count, 0)
 
         # First evaluation
-        evaluator.evaluate_if_necessary(t=3, episodes=3, env=env, agent=agent)
-        self.assertEqual(agent.act.call_count, self.n_runs)
-        self.assertEqual(agent.stop_episode.call_count, self.n_runs)
+        agent_evaluator.evaluate_if_necessary(
+            t=3, episodes=3, env=env, agent=agent)
+        self.assertEqual(agent.act.call_count, self.n_episodes)
+        self.assertEqual(agent.stop_episode.call_count, self.n_episodes)
         if self.save_best_so_far_agent:
             self.assertEqual(agent.save.call_count, 1)
         else:
             self.assertEqual(agent.save.call_count, 0)
 
         # Second evaluation with the same score
-        evaluator.evaluate_if_necessary(t=6, episodes=6, env=env, agent=agent)
-        self.assertEqual(agent.act.call_count, 2 * self.n_runs)
-        self.assertEqual(agent.stop_episode.call_count, 2 * self.n_runs)
+        agent_evaluator.evaluate_if_necessary(
+            t=6, episodes=6, env=env, agent=agent)
+        self.assertEqual(agent.act.call_count, 2 * self.n_episodes)
+        self.assertEqual(agent.stop_episode.call_count, 2 * self.n_episodes)
         if self.save_best_so_far_agent:
             self.assertEqual(agent.save.call_count, 1)
         else:
             self.assertEqual(agent.save.call_count, 0)
 
         # Third evaluation with a better score
         env.step.return_value = ('obs', 1, True, {})
-        evaluator.evaluate_if_necessary(t=9, episodes=9, env=env, agent=agent)
-        self.assertEqual(agent.act.call_count, 3 * self.n_runs)
-        self.assertEqual(agent.stop_episode.call_count, 3 * self.n_runs)
+        agent_evaluator.evaluate_if_necessary(
+            t=9, episodes=9, env=env, agent=agent)
+        self.assertEqual(agent.act.call_count, 3 * self.n_episodes)
+        self.assertEqual(agent.stop_episode.call_count, 3 * self.n_episodes)
         if self.save_best_so_far_agent:
             self.assertEqual(agent.save.call_count, 2)
         else:
             self.assertEqual(agent.save.call_count, 0)
 
 
+@testing.parameterize(
+    *testing.product({
+        'n_episodes': [None, 1],
+        'n_timesteps': [2, 5, 6],
+    })
+)
```

Comment:
`test_needs_reset` does not depend on these parameters, but now it is run 2*3 times because of them. Can you split the class and add `@testing.parameterize` to `test_timesteps` only?